src/mscorlib/corefx/System/Globalization/JapaneseCalendar.Unix.cs

```diff
@@ -1,24 +1,87 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Generic;
+
 namespace System.Globalization
 {
     public partial class JapaneseCalendar : Calendar
     {
-        private static int GetJapaneseEraCount()
+        private static EraInfo[] GetJapaneseEras()
         {
-            //UNIXTODO: Implement this fully.
-            return 0;
+            string[] eraNames;
+            if (!CalendarData.EnumCalendarInfo("ja-JP", CalendarId.JAPAN, CalendarDataType.EraNames, out eraNames))
+            {
+                return null;
+            }
+
+            string[] abbrevEnglishEraNames;
+            if (!CalendarData.EnumCalendarInfo("en", CalendarId.JAPAN, CalendarDataType.AbbrevEraNames, out abbrevEnglishEraNames))
```

Comment:
Because we are looking up the abbreviated English names of the Japanese Eras.  This is needed for the last parameter in the EraInfo constructor.  EraInfo needs the Era Name, Abbreviated Era Name, and the Abbreviated Era Name in English.